examples/step-69/step-69.cc

```diff
@@ -0,0 +1,2893 @@
+            if (!face->at_boundary())
+              continue;
+
+            const auto &fe_face_values = scratch.reinit(cell, f);
+
+            const unsigned int n_face_q_points =
+              fe_face_values.get_quadrature().size();
+
+            for (unsigned int j = 0; j < dofs_per_cell; ++j)
+              {
+                if (!discretization->finite_element.has_support_on_face(j, f))
+                  continue;
+
+                // Note that "normal" will only represent the contributions
+                // from one of the faces in the support of the shape
+                // function phi_j. So we cannot normalize this local
+                // contribution right here, we have to take it "as is",
+                // store it and pass it to the copy data routine. The
+                // proper normalization requires an additional loop on
+                // nodes.
+                Tensor<1, dim> normal;
+                if (id == free_slip)
+                  {
+                    for (unsigned int q = 0; q < n_face_q_points; ++q)
+                      normal += fe_face_values.normal_vector(q) *
+                                fe_face_values.shape_value(j, q);
+                  }
+
+                const auto index = local_dof_indices[j];
+
+                Point<dim> position;
+                const auto global_index = partitioner->local_to_global(index);
+                for (unsigned int v = 0;
+                     v < GeometryInfo<dim>::vertices_per_cell;
+                     ++v)
+                  if (cell->vertex_dof_index(v, 0) == global_index)
+                    position = cell->vertex(v);
+
+                const auto old_id =
+                  std::get<1>(local_boundary_normal_map[index]);
+                local_boundary_normal_map[index] =
+                  std::make_tuple(normal, std::max(old_id, id), position);
+              }
+          }
+      };
+
+      // Last, we provide a copy_local_to_global function as required for
+      // the WorkStream
+      const auto copy_local_to_global = [&](const auto &copy) {
+        const auto &is_artificial             = copy.is_artificial;
+        const auto &local_dof_indices         = copy.local_dof_indices;
+        const auto &local_boundary_normal_map = copy.local_boundary_normal_map;
+        const auto &cell_lumped_mass_matrix   = copy.cell_lumped_mass_matrix;
+        const auto &cell_cij_matrix           = copy.cell_cij_matrix;
+
+        if (is_artificial)
+          return;
+
+        for (const auto &it : local_boundary_normal_map)
+          {
+            auto &normal   = std::get<0>(boundary_normal_map[it.first]);
+            auto &id       = std::get<1>(boundary_normal_map[it.first]);
+            auto &position = std::get<2>(boundary_normal_map[it.first]);
+
+            const auto &new_normal   = std::get<0>(it.second);
+            const auto &new_id       = std::get<1>(it.second);
+            const auto &new_position = std::get<2>(it.second);
```

Comment:
All 6 of these aliases are single-use -- just put their initializers in the places where they're used below.